### PR TITLE
add p2p-webrtc-direct pattern

### PR DESCRIPTION
**Pull request by backkem**
This PR ports the `p2p-webrtc-direct` pattern from `js-mafmt` in preparation of a `go-libp2p-webrtc-direct` transport. It depends on:
- [x] multiformats/go-multiaddr#93
- [x] multiformats/go-multiaddr-dns#11

I'll probably need some help with the dependency management.

### CODE REVIEW

File: patterns.go [hunk 1]
```diff
@@ -4,6 +4,25 @@ import (
...
+	madns "github.com/multiformats/go-multiaddr-dns"
+)
+
+// Define a dns4 format multiaddr
+var DNS4 = Base(madns.P_DNS4)
+
+// Define a dns6 format multiaddr
+var DNS6 = Base(madns.P_DNS6)
+
+var _DNS = Or(
```
Comment (raulk) [inline patterns.go]: If this is copied from js-mafmt, the _ convention usually denotes unexported vars. In go you can lowercase this.

File: patterns.go [hunk 2]
```diff
@@ -4,6 +4,25 @@ import (
...
+	Base(madns.P_DNSADDR),
+	DNS4,
+	DNS6,
+)
+
+// Define a dns4 or dns6 format multiaddr
+var DNS = Or(
+	And(_DNS, Base(ma.P_TCP)),
```
Comment (raulk) [inline patterns.go]: This one is wonky at origin: https://github.com/multiformats/js-mafmt/issues/40

**Review by raulk — CHANGES_REQUESTED** [review]
Minor comments. Overall LGTM, although I'm worried about turning mafmt into a *-import module in the long run.

To avoid stalling, I'll commit the changes myself on your branch.

EDIT: I can't because this repo is under whyrusleeping's account rather than in the multiformats org. I'll clone it there, migrate downstream usages in our control, and port over this PR.

**Comment by backkem** [comment]
This was added in multiformats/go-multiaddr-fmt.

**Pull request closed without merging**